File: app/src/main/java/com/glmiyamoto/learningkotlin/java/Collections.java
package com.glmiyamoto.learningkotlin.java;

import java.util.ArrayList;
import java.util.List;

public class Collections {
    void test() {
        final List<Test> list = new ArrayList<>();
        list.add(new Test(0, "Test 0"));
        list.add(new Test(1, "Test 1"));
        list.add(new Test(2, "Test 2"));
        list.add(new Test(3, "Test 3"));
        list.add(new Test(4, "Test 4"));
        list.add(new Test(5, "Test 5"));

        Test first = null;
        for (Test item : list) {
            if (item.getId() >= 3 && item.getId() <= 5) {
                first = item;
                break;
            }
        }

        Test last = null;
        for (int i = list.size() - 1; i >= 0; i--) {
            Test item = list.get(i);
            if (item.getId() >= 3 && item.getId() <= 5) {
                last = item;
                break;
            }
        }

        List<Test> even = new ArrayList<>();
        for (Test item : list) {
            if (item.getId() % 2 == 0)
                even.add(item);
        }

        List<Integer> listId = new ArrayList<>();
        for (Test item : list) {
            listId.add(item.getId());
        }
    }

    class Test {
        private final int mId;
        private String mName;

        public Test(int id, String name) {
            mId = id;
            mName = name;
        }

        public int getId() {
            return mId;
        }

        public String getName() {
            return mName;
        }

        public void setName(String name) {
            mName = name;
        }
    }
}

File: app/src/main/java/com/glmiyamoto/learningkotlin/kotlin/Collections.kt
package com.glmiyamoto.learningkotlin.kotlin

class Collections {
    fun test() {
        val list = listOf(
                Test(0, "Test 0"),
                Test(1, "Test 1"),
                Test(2, "Test 2"),
                Test(3, "Test 3"),
                Test(4, "Test 4"),
                Test(5, "Test 5")
        )

        val first = list.firstOrNull { it.id in 3..5 }

        val last = list.lastOrNull { it.id in 3..5 }

        val even = list.filter { it.id % 2 == 0 }

        val listId = list.map { it.id }

        val filterNotNull = list.filterNotNull()
        val groupBy = list.groupBy { it.id }
        val distinctBy = list.distinctBy { it.name }
        val sortedBy = list.sortedBy { it.id }
        val sumBy = list.sumBy { it.id }
        val maxBy = list.maxBy { it.id }
        val minBy = list.minBy { it.id }

        val empty = list.isEmpty() || list.isNotEmpty()

        list.forEach { it.name += " ${it.id}" }
    }

    inner class Test(val id: Int, var name: String)
}